tests/test_osv_tools.py

```
@@ -0,0 +1,73 @@
+    ]
+    monkeypatch.setattr(httpx, "Client", lambda **_kwargs: _Client(responses, calls))
+    osv_tools._fetch_osv.cache_clear()
+
+    assert osv_tools._osv_lookup_impl("CVE-2026-1", "1.0").startswith("Error: OSV lookup failed")
+    assert "NOT AFFECTED" in osv_tools._osv_lookup_impl("CVE-2026-1", "1.0")
+    assert len(calls) == 2
+
+
+@pytest.mark.asyncio
+async def test_async_lookup_does_not_block_event_loop(monkeypatch) -> None:
+    def slow_lookup(*_args) -> str:
```

Ruff (0.16.1)

[warning] Missing type annotation for `*_args`

(ANN002)